### ISSUE TITLE
Cannot Scan on Android but works on IOS [BUILD RELEASE]

### ISSUE THREAD

**Issue by loicgeek** [juliuscanute/qr_code_scanner#141]
* Device:
` Galaxy Note9 Android 10`
* Android Manifest
```
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="com.chairwatch.chairwatch_flutter">
    <!-- io.flutter.app.FlutterApplication is an android.app.Application that
         calls FlutterMain.startInitialization(this); in its onCreate method.
         In most cases you can leave this as-is, but you if you want to provide
         additional functionality it is fine to subclass or reimplement
         FlutterApplication and put your custom class here. -->
    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
    <uses-permission android:name="android.permission.INTERACT_ACROSS_USERS_FULL"/>
    <uses-permission android:name="android.permission.READ_CALENDAR" />
    <uses-permission android:name="android.permission.WRITE_CALENDAR" />

    <application
        android:name=".Application"
        android:label="@string/app_name"
        android:icon="@mipmap/ic_launcher"
        android:allowBackup="false"
        android:fullBackupContent="false"
        android:hardwareAccelerated="true"
        >
        <activity
            android:name=".MainActivity"
            android:launchMode="singleTop"
            android:theme="@style/LaunchTheme"
            android:configChanges="orientation|keyboardHidden|keyboard|screenSize|smallestScreenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
            android:hardwareAccelerated="true"
            android:windowSoftInputMode="adjustResize">
            <intent-filter>
                <action android:name="android.intent.action.MAIN"/>
                <category android:name="android.intent.category.LAUNCHER"/>
            </intent-filter>

              <!-- Add below to ensure we get the payload when tapping on a notification -->
        <intent-filter>
            <action android:name="FLUTTER_NOTIFICATION_CLICK" />
            <category android:name="android.intent.category.DEFAULT" />
        </intent-filter>

        <!--deep linking config  -->
        <intent-filter>
                <action android:name="android.intent.action.VIEW" />
                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />
                <data
                    android:scheme="cw"
                    android:host="chairwatch.com" />
        </intent-filter>
        </activity>
        <activity
    android:name="com.yalantis.ucrop.UCropActivity"
    android:screenOrientation="portrait"
    android:theme="@style/Theme.AppCompat.Light.NoActionBar"/>



    <!-- Set custom default icon. This is used when no icon is set for incoming notification messages. -->
<meta-data
    android:name="com.google.firebase.messaging.default_notification_icon"
    android:resource="@drawable/logo_big_grey" />
<!-- Set color used with incoming notification messages. This is used when no color is set for the incoming notification message.  -->
<meta-data
    android:name="com.google.firebase.messaging.default_notification_color"
    android:resource="@color/black" />
        



        <!-- Fcebook config-->

    <meta-data android:name="com.facebook.sdk.ApplicationId" 
        android:value="@string/facebook_app_id"/>
    
    <activity android:name="com.facebook.FacebookActivity"
        android:configChanges=
                "keyboard|keyboardHidden|screenLayout|screenSize|orientation"
        android:label="@string/app_name" />
    <activity
        android:name="com.facebook.CustomTabActivity"
        android:exported="true">
        <intent-filter>
            <action android:name="android.intent.action.VIEW" />
            <category android:name="android.intent.category.DEFAULT" />
            <category android:name="android.intent.category.BROWSABLE" />
            <data android:scheme="@string/fb_login_protocol_scheme" />
        </intent-filter>
    </activity>

            <!-- end facebook config-->
         <meta-data
        android:name="com.google.android.geo.API_KEY"
        android:value="@string/google_map_api_key"/>

        <provider
            android:name="com.pichillilorenzo.flutter_inappwebview.InAppWebViewFileProvider"
            android:authorities="${applicationId}.flutter_inappwebview.fileprovider"
            android:exported="false"
            android:grantUriPermissions="true">
            <meta-data
                android:name="android.support.FILE_PROVIDER_PATHS"
                android:resource="@xml/provider_paths" />
        </provider>

        <!-- Don't delete the meta-data below.
             This is used by the Flutter tool to generate GeneratedPluginRegistrant.java -->
        <meta-data
            android:name="flutterEmbedding"
            android:value="2" />
    </application>
</manifest>
```

* Error
```
Fatal Exception: java.lang.RuntimeException: java.lang.NoSuchMethodException: d.d.d.e.values []
       at java.lang.Enum.enumValues(Enum.java:270)
       at java.lang.Enum.access$000(Enum.java:61)
       at java.lang.Enum$1.create(Enum.java:277)
       at java.lang.Enum$1.create(Enum.java:275)
       at libcore.util.BasicLruCache.get(BasicLruCache.java:63)
       at java.lang.Enum.getSharedConstants(Enum.java:289)
       at java.lang.Class.getEnumConstantsShared(Class.java:2428)
       at java.util.EnumMap.getKeyUniverse(EnumMap.java:755)
       at java.util.EnumMap.<init>(EnumMap.java:138)
       at com.journeyapps.barcodescanner.A.a(:4)
       at com.journeyapps.barcodescanner.BarcodeView.q(:27)
       at com.journeyapps.barcodescanner.BarcodeView.s(:21)
       at com.journeyapps.barcodescanner.BarcodeView.l(:3)
       at com.journeyapps.barcodescanner.l.a(:28)
       at com.journeyapps.barcodescanner.l.p(:48)
       at com.journeyapps.barcodescanner.l.b(:3)
       at com.journeyapps.barcodescanner.g.handleMessage(:13)
       at android.os.Handler.dispatchMessage(Handler.java:103)
       at android.os.Looper.loop(Looper.java:237)
       at android.app.ActivityThread.main(ActivityThread.java:8125)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
```

**Comment by jfdeveloper92** [comment]
@loicgeek did you ever find a solution. having the same problem.

**Comment by 2math** [comment]
It looks like proguard obfuscation problem

**Comment by jlukas99** [comment]
same problem :/

**Comment by jfdeveloper92** [comment]
I was able to get mine to scan. At first it was because I was not on flutter embedding 2. This was when I would click the button nothing would happen. Once I switched over to embedding 2 It seemed to work fine until I released it on the play store. Don't know why but this was because I compiled it on my Windows machine and deployed to the play store. Once I compiled it the same exact way on the Mac it worked fine

**Comment by jlukas99** [comment]
i working on my macbook and still nothing, camera just working but not scanning :/ 

`<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="com.idappstudio.beeloyal4business">

    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.CAMERA" />
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />

    <application
        android:name=".Application"
        android:label="Beeloyal for business"
        android:icon="@mipmap/ic_launcher">
        <activity
            android:name=".MainActivity"
            android:launchMode="singleTop"
            android:theme="@style/LaunchTheme"
            android:configChanges="orientation|keyboardHidden|keyboard|screenSize|smallestScreenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
            android:screenOrientation="portrait"
            android:hardwareAccelerated="true"
            android:windowSoftInputMode="adjustResize">
            
            <meta-data
              android:name="io.flutter.embedding.android.NormalTheme"
              android:resource="@style/NormalTheme"
              />
            
            <meta-data
              android:name="io.flutter.embedding.android.SplashScreenDrawable"
              android:resource="@drawable/launch_background"
              />
            <intent-filter>
                <action android:name="android.intent.action.MAIN"/>
                <category android:name="android.intent.category.LAUNCHER"/>
            </intent-filter>

            <intent-filter>
                <action android:name="FLUTTER_NOTIFICATION_CLICK" />
                <category android:name="android.intent.category.DEFAULT" />
            </intent-filter>
        </activity>
        <meta-data
            android:name="flutterEmbedding"
            android:value="2" />
    </application>
</manifest>
`

**Comment by jfdeveloper92** [comment]
@jlukas99  What happens if you remove the .application from your manifest

**Comment by jfdeveloper92** [comment]
@jlukas99 Also. I see that you are using notifications. Im guessing from the looks of it its firebase messaging. Have you tested  messaging on Android since migrating to Embedding 2. I had an issue on Android where it would cause the app to crash on physical devices (Did not test emulator) whenever the notification was sent.

**Comment by jlukas99** [comment]
Still nothing :c, yea notifications working :)

`I/CameraManager(30430): ;intelligent-mode=-1;iso=auto;iso-values=auto,100,200,400,800;jpeg-quality=96;jpeg-thumbnail-height=384;jpeg-thumbnail-quality=100;jpeg-thumbnail-size-values=512x384,512x288,384x384,320x240,0x0;jpeg-thumbnail-width=512;max-exposure-compensation=20;max-num-detected-faces-hw=16;max-num-detected-faces-sw=0;max-num-focus-areas=1;max-num-metering-areas=0;max-zoom=30;maxaperture-value-denominator=100;maxaperture-value-numerator=153;metering=center;metering-areas=;metering-values=matrix,center,spot;min-exposure-compensation=-20;odc=false;phase-af=off;phase-af-values=off;picture-format=jpeg;picture-format-values=jpeg;picture-size=4128x3096;picture-size-values=4128x3096,4128x2322,3264x2448,3264x1836,3088x3088,2160x2160,2048x1536,2048x1152,1920x1080,1440x1080,1280x720,960x720,640x480,320x240;preferred-preview-size-for-video=1920x1080;preview-format=yuv420sp;preview-format-values=yuv420sp,yuv420p;preview-fps-range=15000,30000;preview-fps-range-values=(15000,15000),(24000,24000),(15000,30000),(30000,30000);preview-frame-rate=30;preview-frame-rate-values=15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30;preview-size=1920x1080;preview-size-values=1920x1080,1440x1080,1072x1072,1280x720,1056x704,960x720,880x720,720x720,720x480,640x480,352x288,320x240,176x144;rotation=0;rt-hdr=off;rt-hdr-values=off;saturation=0;saturation-max=2;saturation-min=-2;scene-mode=auto;scene-mode-values=auto;sharpness=0;sharpness-max=2;sharpness-min=-2;smooth-zoom-supported=false;vertical-view-angle=39.4;video-frame-format=nv21;video-size=1920x1080;video-size-values=1920x1080,1440x1080,1280x720,960x720,800x450,720x480,640x480,480x320,352x288,320x240,176x144;video-snapshot-supported=true;video-stabilization-supported=false;vrmode=-1;vrmode-supported=true;wb-level=-1;wdr=0;whitebalance=auto;whitebalance-values=auto,incandescent,fluorescent,daylight,cloudy-daylight;zoom=0;zoom-ratios=100,104,109,114,120,125,131,138,144,151,158,166,174,182,190,200,209,219,229,240,251,263,276,289,303,317,332,348,364,381,400;zoom-supported=true
I/CameraConfiguration(30430): Requesting focus mode value from among: [auto]
I/CameraConfiguration(30430): Supported focus mode values: [auto, macro, continuous-video, continuous-picture]
I/CameraConfiguration(30430): Can set focus mode to: auto
I/CameraConfiguration(30430): Focus mode already set to auto
I/CameraConfiguration(30430): Requesting flash mode value from among: [off]
I/CameraConfiguration(30430): Supported flash mode values: [off, auto, on, torch]
I/CameraConfiguration(30430): Can set flash mode to: off
I/CameraConfiguration(30430): Flash mode already set to off
I/PreviewScalingStrategy(30430): Viewfinder size: 1544x886
I/PreviewScalingStrategy(30430): Preview in order of preference: [1920x1080, 1280x720, 1440x1080, 1056x704, 960x720, 720x480, 880x720, 1072x1072, 640x480, 720x720, 352x288, 320x240, 176x144]
I/CameraManager(30430): Final camera parameters: 3dnr=false;Infinity=Infinity;antibanding=50hz;antibanding-values=auto,50hz;auto-exposure-lock-supported=true;auto-whitebalance-lock-supported=true;best-capture=0;brightness=0;brightness-max=2;brightness-min=-2;burst-capture=0;burstshot-fps-values=(5,5);constant-growth-rate-zoom-supported=true;contrast=auto;drc=false;dual_mode=-1;dualrecording-hint=-1;dynamic-range-control=off;effect=none;effect-available-fps-values=(10000,24000);effect-values=none;effectrecording-hint=0;exposure-compensation=0;exposure-compensation-step=0.1;fast-fps-mode=-1;flash-mode=off;flash-mode-values=off,auto,on,torch;fnumber-value-denominator=10;fnumber-value-numerator=17;focal-length=3.71;focallength-35mm-value=27;focallength-value-denominator=100;focallength-value-numerator=371;focus-areas=(0,0,0,0,0);focus-distances=0.10,1.20,Infinity;focus-mode=auto;focus-mode-values=auto,macro,continuous-video,continuous-picture;hdr-mode=0;horizontal-view-angle=62.2;hue=0;hue-max=2;hue-min=-2;imageuniqueid-value=X13LSKA00MM X13LSMD01MA
I/CameraManager(30430): ;intelligent-mode=-1;iso=auto;iso-values=auto,100,200,400,800;jpeg-quality=96;jpeg-thumbnail-height=384;jpeg-thumbnail-quality=100;jpeg-thumbnail-size-values=512x384,512x288,384x384,320x240,0x0;jpeg-thumbnail-width=512;max-exposure-compensation=20;max-num-detected-faces-hw=16;max-num-detected-faces-sw=0;max-num-focus-areas=1;max-num-metering-areas=0;max-zoom=30;maxaperture-value-denominator=100;maxaperture-value-numerator=153;metering=center;metering-areas=;metering-values=matrix,center,spot;min-exposure-compensation=-20;odc=false;phase-af=off;phase-af-values=off;picture-format=jpeg;picture-format-values=jpeg;picture-size=4128x3096;picture-size-values=4128x3096,4128x2322,3264x2448,3264x1836,3088x3088,2160x2160,2048x1536,2048x1152,1920x1080,1440x1080,1280x720,960x720,640x480,320x240;preferred-preview-size-for-video=1920x1080;preview-format=yuv420sp;preview-format-values=yuv420sp,yuv420p;preview-fps-range=15000,30000;preview-fps-range-values=(15000,15000),(24000,24000),(15000,30000),(30000,30000);preview-frame-rate=30;preview-frame-rate-values=15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30;preview-size-values=1920x1080,1440x1080,1072x1072,1280x720,1056x704,960x720,880x720,720x720,720x480,640x480,352x288,320x240,176x144;rotation=0;rt-hdr=off;rt-hdr-values=off;saturation=0;saturation-max=2;saturation-min=-2;scene-mode=auto;scene-mode-values=auto;sharpness=0;sharpness-max=2;sharpness-min=-2;smooth-zoom-supported=false;vertical-view-angle=39.4;video-frame-format=nv21;video-size=1920x1080;video-size-values=1920x1080,1440x1080,1280x720,960x720,800x450,720x480,640x480,480x320,352x288,320x240,176x144;video-snapshot-supported=true;video-stabilization-supported=false;vrmode=-1;vrmode-supported=true;wb-level=-1;wdr=0;whitebalance=auto;whitebalance-values=auto,incandescent,fluorescent,daylight,cloudy-daylight;zoom=0;zoom-ratios=100,104,109,114,120,125,131,138,144,151,158,166,174,182,190,200,209,219,229,240,251,263,276,289,303,317,332,348,364,381,400;zoom-supported=true;preview-size=1920x1080
I/CenterCropStrategy(30430): Preview: 1080x1920; Scaled: 886x1575; Want: 886x1544
I/CameraPreview(30430): Starting preview
D/CameraInstance(30430): Starting preview
W/MapperHal(30430): buffer descriptor with invalid usage bits 0x2000
W/MapperHal(30430): buffer descriptor with invalid usage bits 0x2000
W/MapperHal(30430): buffer descriptor with invalid usage bits 0x2000
W/MapperHal(30430): buffer descriptor with invalid usage bits 0x2000
W/MapperHal(30430): buffer descriptor with invalid usage bits 0x2000
W/MapperHal(30430): buffer descriptor with invalid usage bits 0x2000
W/MapperHal(30430): buffer descriptor with invalid usage bits 0x2000
W/MapperHal(30430): buffer descriptor with invalid usage bits 0x2000
W/MapperHal(30430): buffer descriptor with invalid usage bits 0x2000
W/MapperHal(30430): buffer descriptor with invalid usage bits 0x2000
W/MapperHal(30430): buffer descriptor with invalid usage bits 0x2000
W/MapperHal(30430): buffer descriptor with invalid usage bits 0x2000
I/AutoFocusManager(30430): Current focus mode 'auto'; use auto focus? true
V/FA      (30430): Inactivity, disconnecting from the service
I/eloyal4busines(30430): Background concurrent copying GC freed 3579(448KB) AllocSpace objects, 10(11MB) LOS objects, 50% free, 8MB/17MB, paused 261us total 114.124ms
I/eloyal4busines(30430): Background concurrent copying GC freed 3606(432KB) AllocSpace objects, 10(11MB) LOS objects, 49% free, 8MB/16MB, paused 449us total 102.376ms
I/eloyal4busines(30430): Background concurrent copying GC freed 3576(432KB) AllocSpace objects, 10(11MB) LOS objects, 49% free, 8MB/17MB, paused 5.106ms total 89.116ms
I/eloyal4busines(30430): Waiting for a blocking GC ProfileSaver
I/eloyal4busines(30430): WaitForGcToComplete blocked ProfileSaver on HeapTrim for 12.044ms
`

**Comment by jfdeveloper92** [comment]
@jlukas99 what pack version are you using?

**Comment by jlukas99** [comment]
qr_code_scanner: ^0.0.13

**Comment by jfdeveloper92** [comment]
can you remove the async. I do not have that in mine.

void _onQRViewCreated(QRViewController controller) {
    this._controller = controller;

    controller.scannedDataStream.listen((scanData) {
      if (!_barcodeFound) {
        if (scanData.length > 0) {
          print("scanned: $scanData");
          controller.pauseCamera();
          setState(() {
            _barcodeFound = true;
          });
          Navigator.pop(context, scanData);
        }
      }
    });
  }

**Comment by jfdeveloper92** [comment]
> Still nothing :c, yea notifications working :)
> 
> https://hastebin.com/enizuquwam.apache

Also about the notifications it only crashed if the app was closed and dismissed from recents

**Comment by juliansteenbakker** [comment]
I tried it on a Galaxy S9+ and everything worked fine. I will be upgrading this plugin to embedding v2 #132 somewhere this week so maybe that will fix the problem. I'll let you know when it gets released.

**Comment by juliansteenbakker** [comment]
I've released v0.1.0 which uses embedding V2. Please check if your problem still exists.
